### PR TITLE
More System.Text.Json updates for .NET 9

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #42592.

- Read multiple JSON documents using Utf8JsonReader.
- Indentation options.
- DeepEquals for JsonElement and JsonNode.
- Update code snippets to use Web options singleton.
- Add links from What's new.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/whats-new/dotnet-9/libraries.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/core/whats-new/dotnet-9/libraries.md) | [docs/core/whats-new/dotnet-9/libraries](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-9/libraries?branch=pr-en-us-43100) |
| [docs/standard/serialization/system-text-json/extract-schema.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/standard/serialization/system-text-json/extract-schema.md) | [docs/standard/serialization/system-text-json/extract-schema](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/extract-schema?branch=pr-en-us-43100) |
| [docs/standard/serialization/system-text-json/handle-overflow.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/standard/serialization/system-text-json/handle-overflow.md) | [docs/standard/serialization/system-text-json/handle-overflow](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/handle-overflow?branch=pr-en-us-43100) |
| [docs/standard/serialization/system-text-json/how-to.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/standard/serialization/system-text-json/how-to.md) | [docs/standard/serialization/system-text-json/how-to](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to?branch=pr-en-us-43100) |
| [docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/standard/serialization/system-text-json/migrate-from-newtonsoft.md) | [docs/standard/serialization/system-text-json/migrate-from-newtonsoft](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?branch=pr-en-us-43100) |
| [docs/standard/serialization/system-text-json/use-dom.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/standard/serialization/system-text-json/use-dom.md) | [docs/standard/serialization/system-text-json/use-dom](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-dom?branch=pr-en-us-43100) |
| [docs/standard/serialization/system-text-json/use-utf8jsonreader.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/standard/serialization/system-text-json/use-utf8jsonreader.md) | [docs/standard/serialization/system-text-json/use-utf8jsonreader](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-utf8jsonreader?branch=pr-en-us-43100) |
| [docs/standard/serialization/system-text-json/visual-basic-support.md](https://github.com/dotnet/docs/blob/2a530e3c469a0cc1f6a8629b8ab7051e2506963a/docs/standard/serialization/system-text-json/visual-basic-support.md) | [docs/standard/serialization/system-text-json/visual-basic-support](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/visual-basic-support?branch=pr-en-us-43100) |

### CODE REVIEW

File: docs/standard/serialization/system-text-json/snippets/how-to-contd/vb/Program.vb [hunk 1]
```diff
@@ -14,49 +14,49 @@
             ImmutableTypes.Program.Main()
             Console.WriteLine()
 
-            Console.WriteLine("======== Field support =========")
-            Fields.Program.Main()
-            Console.WriteLine()
+            'Console.WriteLine("======== Field support =========")
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/snippets/how-to-contd/vb/Program.vb:17]: Why is this being commented out?

File: docs/standard/serialization/system-text-json/snippets/how-to/csharp/JsonDocumentDataAccess.cs [hunk 2]
```diff
@@ -70,5 +74,15 @@ private static void AverageGrades_Alternative(string jsonString)
             Console.WriteLine($"Average grade : {average}");
             // </AverageGrades2>
         }
+
+        private static void Compare()
+        {
+            // <DeepEquals>
+            JsonElement left = JsonDocument.Parse("10e-3").RootElement;
+            JsonElement right = JsonDocument.Parse("0.01").RootElement;
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/snippets/how-to/csharp/JsonDocumentDataAccess.cs:82]: This is a mistake I introduced, should be:
```suggestion
            JsonElement right = JsonDocument.Parse("0.001").RootElement;
```

File: docs/standard/serialization/system-text-json/snippets/how-to/csharp/Program.cs [hunk 3]
```diff
@@ -6,145 +6,145 @@ public class Program
 {
     static async Task Main(string[] args)
     {
-        Console.WriteLine("\n============================= Roundtrip to DateOnly and TimeOnly props.\n");
-        RoundtripDateOnly.Run();
+        //Console.WriteLine("\n============================= Roundtrip to DateOnly and TimeOnly props.\n");
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/snippets/how-to/csharp/Program.cs:9]: Same question here, is this accidental?

**Review by eiriktsarpalis — APPROVED** [review]

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

Once you address the open comments, let's :shipit:

File: docs/standard/serialization/system-text-json/snippets/how-to-contd/vb/Program.vb (hunk 1 above)
Comment (gewarren) [reply docs/standard/serialization/system-text-json/snippets/how-to-contd/vb/Program.vb:17]: This code isn't shown in the docs anywhere. It's just a way to run the different snippets/methods. I didn't want to have to run all of them - just the one I was adding.

File: docs/standard/serialization/system-text-json/snippets/how-to/csharp/Program.cs (hunk 3 above)
Comment (gewarren) [reply docs/standard/serialization/system-text-json/snippets/how-to/csharp/Program.cs:9]: This also isn't shown in the docs.

**Pull request merged**